Packs/Veeam/Integrations/VeeamONERESTAPI/VeeamONERESTAPI.py

```diff
@@ -0,0 +1,629 @@
+        return result
+
+    elif command == 'fetch-incidents':
+        max_triggered_alarms_for_fetch = int(params.get('triggered_alarms_per_request', MAX_EVENTS_FOR_FETCH))
+
+        next_run, incidents = fetch_incidents(
+            client=client,
+            last_run=demisto.getLastRun(),  # getLastRun() gets the last run dict
+            first_fetch_time=datetime.strftime(first_fetch_time, DATE_FORMAT),
+            max_triggered_alarms_for_fetch=max_triggered_alarms_for_fetch)
+
+        demisto.setLastRun(next_run)
+        demisto.incidents(incidents)
+
+    elif command in commands:
+        result = handle_command_with_token_refresh(commands[command], {'client': client, 'args': args}, client, max_attempts)
+        return result
+    else:
+        raise NotImplementedError(f'Command {command} is not implemented.')
+
+
+def get_api_key(client: Client) -> str:
+    credentials: Dict[str, str] = demisto.params().get('credentials')
+    username: str = credentials.get('identifier', '')
+    password: str = credentials.get('password', '')
+    token = update_token(client, username, password)
+    api_key = 'Bearer ' + token
+    return api_key
+
+
+def set_api_key(client: Client, api_key: str) -> None:
+    headers = client.get_headers()
+    headers['Authorization'] = api_key
+    client.set_headers(headers)
+
+
+def handle_command_with_token_refresh(command: Callable, command_params: dict, client: Client, max_attempts: int = MAX_ATTEMPTS):
+    attempts = 0
+
+    while attempts < max_attempts:
+        try:
+            context = demisto.getIntegrationContext()
+            api_key = context.get('token')
+            if not api_key:
+                api_key = get_api_key(client)
+                demisto.setIntegrationContext({'token': api_key})
+
+            set_api_key(client, api_key)
+
+            response = client.get_about_request()
+            version = response.get('version', '')
+            check_version(version)
+
+            res = command(**command_params)
+            return res
+        except Exception as e:
+            status_code = getattr(getattr(e, 'res', None), 'status_code', None)
+            if status_code == HTTPStatus.UNAUTHORIZED:
+                attempts += 1
+                context = demisto.getIntegrationContext()
+                context['token'] = None
+                demisto.setIntegrationContext(context)
+            else:
+                raise e
+
+    else:
+        raise ValueError('Failed to obtain a valid API Key after 3 attempts')
+
+
+def main() -> None:
+
+    params: Dict[str, Any] = demisto.params()
+    args: Dict[str, Any] = demisto.args()
+    url: str = params.get('url', '')
+    verify_certificate: bool = not params.get('insecure', False)
+    proxy: bool = params.get('proxy', False)
+
+    first_fetch_time = arg_to_datetime(
+        arg=params.get('first_fetch', '3 days'),
+        arg_name='First fetch time',
+        required=True
+    )
+    assert first_fetch_time
```

Comment:
If `arg_to_datetime` returns `None` the integration will break.
Please add a more graceful handling here and insure that a valid default value will be set in case of a `None` response, Something like:
```
if not first_fetch_datetime:
    first_fetch_time = datetime.now()
```

Reply:
done